src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +789,234 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			// Non standalone scripts in the after position, of type async or defer, are usually delayed.
+			$strategy = $this->get_intended_strategy( $handle );
+			if ( $this->is_non_blocking_strategy( $strategy ) &&
+				$this->has_non_standalone_inline_script( $handle, 'after' )
+			) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Normalize the data inside the $args parameter and support backward compatibility.
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default false.
+	 *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+	 * }
+	 * @return array        Normalized $args array.
+	 */
+	private function get_normalized_script_args( $handle, $args = array() ) {
```

Comment:
Excellent! noting this will need a `_` prefix for core (convention for private functions)

Reply:
@adamsilverstein I think we need to seek clarity around this, as when searching the contents of `wp-includes` there are _far_ more instances of private function declarations _without_ the `_` preceding the function name than there are _with_ the `_`, by a large amount.

I also can not find anything about this in the official coding standards, see link [here](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/).

I'm happy to add it, as it changes little aside from having to update one or two references to the function, but I do feel that this is an area of preference as opposed to standard. 

If I am wrong, please let me know :)

Reply:
Since this is a private class method, the prefix is unnecessary. In `wp-includes` at the moment, only 15 of the 114 private methods start with an underscore. The underscore prefix is more commonly used for global functions.